Tests/FluentAssertions.Specs/Exceptions/AsyncFunctionExceptionAssertionSpecs.cs

```diff
@@ -935,6 +935,20 @@ public async Task When_async_method_throws_aggregate_exception_containing_expect
         await action.Should().NotThrowAsync();
     }
 
+    [Fact]
+    public async Task When_async_method_throws_any_exception_it_should_succeed()
```

Comment:
🔧We try to use a more fact-based naming convention these days and avoid words like _should_ and _when_. So in this case `Succeeds_for_any_thrown_exception`. We also started to group the test cases in nested classes named like the API. See for example `CollectionAssertionSpecs`. There we even used partial classes to keep things organized.

Reply:
Yeah, I'm afraid I've sent you into a rabbit hole. The idea is that we group the tests under the nested classes and use names to represent them based on API or capability, e.g. `NotThrowAfterAsync`, `Actions` or `NogGenericAsyncFunctions` (without the `Assertions` part).  And if we _do_ decide to refactor such a big class we either do that
1. In a separate PR
2. In a separate commit if your PR has been properly organized into focused commits we want to keep after merging. 

In your case, we will squash the PR as you have merged the `main` branch into the feature branch. We only [do rebases](https://learngitbranching.js.org/?NODEMO&command=level%20intro4).

So let's remove these nested classed you've added so we can look at this at a later point of time.